src/compiler/checker.ts

```diff
@@ -23917,6 +23918,10 @@ namespace ts {
                     return getTypeFacts(leftType) & TypeFacts.Falsy ?
                         getUnionType([removeDefinitelyFalsyTypes(leftType), rightType], UnionReduction.Subtype) :
                         leftType;
+                case SyntaxKind.QuestionQuestionToken:
+                    return getTypeFacts(leftType) & TypeFacts.EQUndefinedOrNull ?
+                        getUnionType([getNonNullableType(leftType), rightType], UnionReduction.Subtype) :

```

Comment:
Does it make sense to also return a union type for when `strictNullChecks` is off?

Reply:
I'd like to say yes, every value could be null/undefined if strictNullChecks is off and we cannot detect that.

same as:
```ts
declare const a: string
let b = a || 1 // b is string || number
```